lib/src/events/CommandsObserver.test.ts

```diff
@@ -1,14 +1,15 @@
-import { CommandsObserver } from './CommandsObserver';
+import { CommandsObserver, CommandsListener } from './CommandsObserver';
+import { UniqueIdProvider } from '../adapters/UniqueIdProvider';
 
 describe('CommandsObserver', () => {
   let uut: CommandsObserver;
-  let cb1;
-  let cb2;
+  let cb1: CommandsListener;
+  let cb2: CommandsListener;
 
   beforeEach(() => {
     cb1 = jest.fn();
     cb2 = jest.fn();
-    uut = new CommandsObserver();
+    uut = new CommandsObserver(new UniqueIdProvider());
```

Comment:
Because previously `CommandsObserver` was using directly lodash even though there is UniqueIdProvider available. We should reuse services and not create duplicate code.

Here is the diff
![image](https://user-images.githubusercontent.com/12229968/50476086-78bc5700-09cf-11e9-97dc-93656dd4c44f.png)

Reply:
Can we create `UniqueIdProvider` inside the class (in constructor or instantiate in class member declaration) instead of passing it as an argument?

Reply:
That is not good idea for testing. Because if you don't pass dependency like this it is way harder to mock in tests. That is why any of the services shouldn't create their own dependencies. Read first chapter from this: https://nehalist.io/dependency-injection-in-typescript/ there is more reasons why the service should not create it's own dependencies :)

Reply:
Right, but in this specific case - CommandsObserver test doesn't mock `UniqueIdProvider` and doesn't make any verifications on it.